test/functional/legacy/charsearch_spec.lua

```diff
@@ -0,0 +1,42 @@
+-- Test for character searches
+
+local helpers = require('test.functional.helpers')
+local feed, insert = helpers.feed, helpers.insert
+local clear, execute, expect = helpers.clear, helpers.execute, helpers.expect
+
+describe('charsearch', function()
+  setup(clear)
+
+  it('is working', function()
+    insert([[
+      Xabcdefghijkemnopqretuvwxyz
+      Yabcdefghijkemnopqretuvwxyz
+      Zabcdefghijkemnokqretkvwxyz]])
+
+    -- Check that "fe" and ";" work.
+    execute('/^X')
+    feed('ylfep;;p,,p')
+    -- Check that save/restore works.
+    execute('/^Y')
+    feed('ylfep')
+    execute('let csave = getcharsearch()')
+    feed('fip')
+    execute('call setcharsearch(csave)')
+    feed(';p;p')
+    -- Check that setcharsearch() changes the settins.
+    execute('/^Z')
+    feed('ylfep')
+    execute("call setcharsearch({'char': 'k'})")
+    feed(';p')
+    execute("call setcharsearch({'forward': 0})")
+    feed('$;p')
+    execute("call setcharseearch({'until': 1})")
```

Comment:
I didn't even notice this typo in the original test. If I correct it the test fails. 

Is this a vim bug? Could you ask on the vim mailing list (James McCoy wrote the patch 7.4.813 for these functions). Or should we just fix it somehow or leave it?

(I'm not on the vim list that's why I ask.)